### ISSUE TITLE
bug: NAN in latest version of jax when using multimer+bfloat16

### ISSUE THREAD

**Issue by sokrypton** [deepmind/alphafold#724]
I tracked down the issue. 

The issue comes from when the multimer model is run without templates. This msa_mask for the blank appended template is all zeros. This creates a very large negative bias that bfloat16 can't seem to handle.

if you replace:
`bias = (1e9 * (msa_mask - 1.))[:, None, None, :]`
with:
`bias = (1e4 * (msa_mask - 1.))[:, None, None, :]`

No more NANS!

See:
https://github.com/deepmind/alphafold/blob/main/alphafold/model/modules.py#L764

**Comment by sokrypton** [comment]
Another hack is to clip the logits before taking softmax in the Attention Class
replace:
`weights = jax.nn.softmax(logits)`
with:
`weights = jax.nn.softmax(jnp.clip(logits,-1e8,1e8))`

see:
https://github.com/deepmind/alphafold/blob/main/alphafold/model/modules.py#L592
